src/odemis/gui/cont/acquisition.py

```diff
@@ -491,46 +487,52 @@ def __init__(self, tab_data, panel, tab):
         self._panel.txt_cryosecom_est_time.Show()
         self._panel.btn_cryosecom_acqui_cancel.Hide()
         self._panel.Layout()
-        # limit number of characters
-        self._panel.param_Zmin.SetMaxLength(8)
-        self._panel.param_Zmax.SetMaxLength(8)
-        self._panel.param_Zstep.SetMaxLength(8)
-        # set the default values
-        self._panel.param_Zmax.SetValue("10 µm")
-        self._panel.param_Zmin.SetValue("-10 µm")
-        self._panel.param_Zstep.SetValue("1 µm")
 
         # bind events (buttons, checking, ...) with callbacks 
         # for "ACQUIRE" button
         self._panel.btn_cryosecom_acquire.Bind(wx.EVT_BUTTON, self._on_acquire)
         # for "change..." button
         self._panel.btn_cryosecom_change_file.Bind(
-            wx.EVT_BUTTON, self._on_change
+            wx.EVT_BUTTON, self._on_btn_change
         )
         # for "acquire overview" button
         self._panel.btn_acquire_overview.Bind(wx.EVT_BUTTON, self._on_acquire_overview)
         # for "cancel" button
         self._panel.btn_cryosecom_acqui_cancel.Bind(wx.EVT_BUTTON, self._on_cancel)
         # for the check list box
         self._panel.streams_chk_list.Bind(wx.EVT_CHECKLISTBOX, self._on_check_list)
-        # for the z-stack parameters
-        self._panel.param_Zmin.Bind(wx.EVT_TEXT_ENTER, self._on_Zmin_param)
-        self._panel.param_Zmax.Bind(wx.EVT_TEXT_ENTER, self._on_Zmax_param)
-        self._panel.param_Zstep.Bind(wx.EVT_TEXT_ENTER, self._on_Zstep_param)
 
         # callbacks of VA's
         self._tab_data.filename.subscribe(self._on_filename, init=True)
         self._tab_data.zStackActive.subscribe(self._update_zstack_active, init=True)
+        self._tab_data.zMin.subscribe(self._on_z_min, init=True)
+        self._tab_data.zMax.subscribe(self._on_z_max, init=True)
+        self._tab_data.zStep.subscribe(self._on_z_step, init=True)
         self._tab_data.streams.subscribe(self._on_streams, init=True)
         # TODO link .acquiStreams with a callback that is called
         # to check/uncheck items (?)
 
         # VA's connector
-        self._zstack_chkbox_vac = VigilantAttributeConnector(
+        _ = VigilantAttributeConnector(
             va=self._tab_data.zStackActive,
             value_ctrl=self._panel.z_stack_chkbox,
             events=wx.EVT_CHECKBOX,
         )
+        _ = VigilantAttributeConnector(
+            va=self._tab_data.zMin,
+            value_ctrl=self._panel.param_Zmin,
+            events=wx.EVT_COMMAND_ENTER,
+        )
+        _ = VigilantAttributeConnector(
+            va=self._tab_data.zMax,
+            value_ctrl=self._panel.param_Zmax,
+            events=wx.EVT_COMMAND_ENTER,
+        )
+        _ = VigilantAttributeConnector(
+            va=self._tab_data.zStep,
+            value_ctrl=self._panel.param_Zstep,
+            events=wx.EVT_COMMAND_ENTER,
+        )
```

Comment:
I'm really not sure if this might happen. The connections are done in the constructor of the VigilantAttributeConnector class, using references to tab_data and panel which are both members of the Cryo acqui controller class, so the connections should stay working in the scope of the cryo acqui controller. In addition to that, it has been tested several times so far, and the VA connections seem to work fine. So i don't think the python interpreter will destroy these connections.